src/main/java/com/dubture/jenkins/digitalocean/ComputerLauncher.java

```diff
@@ -128,8 +113,16 @@ public void onClosed(Channel channel, IOException cause) {
 
             successful = true;
         } catch (Exception e) {
+            try {
+                Jenkins.getInstance().removeNode(computer.getNode());
+            } catch (Exception ee) {
+                ee.printStackTrace(logger);
+            }
             e.printStackTrace(logger);
         } finally {
+            Date endDate = new Date();
+            logger.println("Done setting up at: " + getUtcDate(endDate));
```

Comment:
not quite sure if it's a good log lvl. Do you really want to print it out all the time?